superdesk/io/feeding_services/__init__.py

```diff
@@ -45,9 +45,40 @@ def _update(self, provider, update):
         :type update: dict
         :return: a list of articles which can be saved in Ingest Collection.
         """
-
         raise NotImplementedError()
 
+    def _test(self, provider):
+        """
+        Subclasses should override this method and do specific config test.
+
+        :param provider: Ingest Provider Details.
+        """
+        return
+
+    def config_test(self, provider=None):
+        """Test provider configuration.
+
+        :param provider: provider data
+        """
+        if not provider:  # nosetests run this for some reason
```

Comment:
because the name `*_test`, I think we should move all tests under `tests` directory and change command like `nosetest tests`